pkg/cli/preview/recorder.go

```diff
@@ -179,11 +200,25 @@ func (r *Recorder) recordReconcileStart(ctx context.Context, u *unstructured.Uns
 func (r *Recorder) recordReconcileEnd(ctx context.Context, u *unstructured.Unstructured, result reconcile.Result, err error) {
 	gknn := gknnFromUnstructured(u)
 
+	if done := r.GKNNDoneReconcile(gknn); done {
+		return
+	}
+
 	info := r.getObjectInfo(gknn)
 	info.events = append(info.events, event{
 		eventType: EventTypeReconcileEnd,
 		object:    u.DeepCopy(),
 	})
+	r.reconcileTrackerMutex.Lock()
+	defer r.reconcileTrackerMutex.Unlock()
+	r.ReconciledResources[gknn] = true
+	r.RemainResourcesCount--
```

Comment:
Nit: can we ever reconcile the same gknn twice?  Might be better to double-check that ReconcileResources[gknn] = false